### PR TITLE
Opi docker custom port

### PR DESCRIPTION
**Pull request by schaefm-ibm**
Thanks for contributing to cloud_controller_ng. To speed up the process of reviewing your pull request please provide us with:

* A short explanation of the proposed change:
Support staging of docker images with Eirini on non-standard port

* An explanation of the use cases your change solves
cf push of docker images with specified EXPOSE port

* Links to any other associated PRs

* [x] I have reviewed the [contributing guide](https://github.com/cloudfoundry/cloud_controller_ng/blob/master/CONTRIBUTING.md)

* [x] I have viewed, signed, and submitted the Contributor License Agreement

* [x] I have made this pull request to the `master` branch

* [x] I have run all the unit tests using `bundle exec rake`

* [x] I have run [CF Acceptance Tests](https://github.com/cloudfoundry/cloud_controller_ng#cf-acceptance-tests-cats)

### CODE REVIEW

File: lib/cloud_controller/opi/stager_client.rb [hunk 1]
```diff
@@ -10,20 +10,54 @@ class StagerClient < BaseClient
     def stage(staging_guid, staging_details)
       logger.info('stage.request', staging_guid: staging_guid)
 
-      if staging_details.lifecycle.type == VCAP::CloudController::Lifecycles::DOCKER
-        complete_staging(staging_guid, staging_details)
-      elsif staging_details.lifecycle.type == VCAP::CloudController::Lifecycles::BUILDPACK
-        staging_request = to_request(staging_guid, staging_details)
-        start_staging(staging_guid, staging_request)
-      else
+      if staging_details.lifecycle.type != VCAP::CloudController::Lifecycles::DOCKER && staging_details.lifecycle.type != VCAP::CloudController::Lifecycles::BUILDPACK
```
Comment (cwlbraa) [inline lib/cloud_controller/opi/stager_client.rb]:
```
unless [VCAP::CloudController::Lifecycles::DOCKER, VCAP::CloudController::Lifecycles::BUILDPACK].include?(staging_details.lifecycle.type)
``` 

would be easier to read here

**Comment by schaefm-ibm** [comment]
@cwlbraa  Followed your recommendation and updated accordingly.

**Comment by julz** [comment]
nag, nag :)

**Comment by hsiliev** [comment]
This seems like a must for running Eirini with external Docker registry. 
Any idea when this can be merged, so it can make its way in kubecf and cf-for-k8s?

**Comment by gdankov** [comment]
Hello @cwlbraa! This PR has been open for a long time. Is there anything else we can do to help with this being merged?

**Comment by cwlbraa** [comment]
hey, I pinged the CAPI team in slack about this, and you should consider doing the same! I'm working on a research project and haven't been keeping up with CAPI CI or Github notifications.

**Comment by aashah** [comment]
Hello,

Feedback was applied. PR looks solid. Rubocop is happy. Tests are green. Let's merge it.

Sorry for the delay.

- Aakash

**Comment by aashah** [comment]
Hello,

We unfortunately had to revert the PR as we found some larger underlying issues with CC on cf-for-k8s. This PR broke staging docker apps as we observed Eirini was not able to communicate back to CC:

```
Post https://TODO:***@capi.cf-system.svc.cluster.local:9023/internal/v3/staging/fde60ac5-8f46-4be7-8fae-19e94aa23751/build_completed?start=true: read tcp 10.0.1.62:51490->10.4.11.87:9023: read: connection reset
```

We wrote a story to address this, and that is currently top of backlog for us - https://www.pivotaltracker.com/story/show/172321999

We will re-merge the PRs alongside completing that story.

- Aakash && @sethboyles

**Pull request merged**